### PR TITLE
[SRBT-256] Onboarding Drawer

### PR DESCRIPTION
**Pull request by Cutaiar**
# [SRBT-256] Onboarding Drawer

- Add `HandleInput` and `OnboardingDrawer` with stories 
- Update radix dialog via shadcn cli
- Adds `@storybook/preview-api` for seamless control of drawer in SB.

# Screenshots

Version 1:

https://github.com/user-attachments/assets/773883e6-4795-4b65-8643-bfa0390a23f7

### CODE REVIEW

File: src/components/profile/widgets/onboarding-drawer/handle-input.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      }
+    };
+
+    return (
+      <div className='relative'>
+        <Input
+          type='text'
+          // TODO: Keep the @ when user types
+          placeholder='@username'
+          className={cn('truncate pl-9 pr-10 shadow-sm', className)}
```
Comment (ChristopherPark3) [inline src/components/profile/widgets/onboarding-drawer/handle-input.tsx:48]: For the shadcn focus borders, here's a good article if you want to change them:

https://github.com/shadcn-ui/ui/issues/1512

File: yarn.lock [hunk 2]
```diff
@@ -19734,6 +19734,7 @@ __metadata:
     "@storybook/addon-essentials": "npm:^8.3.1"
     "@storybook/addon-links": "npm:^8.3.1"
     "@storybook/nextjs": "npm:^8.3.1"
+    "@storybook/preview-api": "npm:^8.3.1"
```
Comment (ChristopherPark3) [inline yarn.lock:19737]: is this anything in particular? or is it just a result of an update

File: src/components/profile/widgets/onboarding-drawer/onboarding-drawer.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof OnboardingDrawer>;
+
+export const Default: Story = {
+  render: (args) => {
+    // This is some SB boiler plate to sync the control with drawer state
+    const [{ open }, updateArgs] = useArgs();
```
Comment (ChristopherPark3) [inline src/components/profile/widgets/onboarding-drawer/onboarding-drawer.stories.tsx:25]: this is sweet im gonna start using this

**Review by ChristopherPark3 — APPROVED** [review]
Looks great! storybook ftw

File: src/components/profile/widgets/onboarding-drawer/handle-input.tsx (hunk 1 above)
Comment (Cutaiar) [reply src/components/profile/widgets/onboarding-drawer/handle-input.tsx:48]: Nice! Yeah I will look into this. Mostly, i want to discuss it with Chris as potential change for the base component.

File: src/components/profile/widgets/onboarding-drawer/onboarding-drawer.stories.tsx (hunk 3 above)
Comment (Cutaiar) [reply src/components/profile/widgets/onboarding-drawer/onboarding-drawer.stories.tsx:25]: Oh yeah! This is the way you do it if you want something in the story to effect the controls too.

File: yarn.lock (hunk 2 above)
Comment (Cutaiar) [reply yarn.lock:19737]: just the result of using the shad cli. Good to be up to date on these things as they usually improve the components.

**Pull request merged**